app/api/logging/pii.py

```diff
@@ -0,0 +1,102 @@
+"""Mask PII from log records.
+
+This module defines a filter that can be attached to a logger to mask PII
+from log records. The filter is applied to all log records, and masks PII
+that looks like social security numbers.
+
+You can add the filter to a handler:
+
+Example:
+    import logging
+    import api.logging.pii as pii
+
+    handler = logging.StreamHandler()
+    handler.addFilter(pii.mask_pii)
+    logger = logging.getLogger(__name__)
+    logger.addHandler(handler)
+
+Or you can add the filter directly to a logger.
+If adding the filter directly to a logger, take note that the filter
+will not be called for child loggers.
+See https://docs.python.org/3/library/logging.html#logging.Logger.propagate
+
+Example:
+    import logging
+    import api.logging.pii as pii
+
+    logger = logging.getLogger(__name__)
+    logger.addFilter(pii.mask_pii)
+"""
+
+import logging
+import re
+from typing import Any, Optional
+
+# Attributes of LogRecord to exclude from the JSON formatted lines. An exclusion list approach is
+# used so that all "extra" attributes can be included in a line.
+EXCLUDE_ATTRIBUTES = {
```

Comment:
Do we want to filter by this list again here? This is the same list as in formatters.py and is more about what is output than where to mask pii.

Reply:
Thanks for the context and for noticing the duplication, I wasn't aware of that. I'll remove it from here!

Reply:
Oh hm I notice I'm actually not using that dictionary anywhere if I remove it from here. Sounds like the intention is to exclude those fields? Is it because it's too noisy?

Reply:
i tried logging a few things in healthcheck as a test and don't see any random attributes from that list even when i completely remove this dictionary. can't seem to figure out what it does.

<img width="1570" alt="image" src="https://user-images.githubusercontent.com/447859/214184106-c795aa07-e2fd-4026-802c-d53e5df2ecd4.png">

i just went ahead and removed it. also, even if we do end up seeing logs with filename, lineno, etc. i don't see much of an issue with that being in the logs

Reply:
I think you're not seeing them because there's also similar code in decodelog that excludes them from the extra part of the output (blue part). So I agree with this change.